modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.replicator;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.ignite.internal.replicator.message.CleanupRequest;
+import org.apache.ignite.internal.replicator.message.ReplicaMessagesFactory;
+import org.apache.ignite.internal.replicator.message.ReplicaRequest;
+import org.apache.ignite.internal.replicator.message.ReplicaResponse;
+import org.apache.ignite.internal.replicator.message.WaiteOperationResultRequest;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.apache.ignite.lang.IgniteStringFormatter;
+
+/**
+ * Replica server.
+ * TODO:IGNITE-17257 Implement Replica server-side logic.
+ */
+public class Replica {
+    /** Replicator network message factory. */
+    private static final ReplicaMessagesFactory REPLICA_MESSAGES_FACTORY = new ReplicaMessagesFactory();
+
+    /** Replica group identity, this id is the same as the considered partition's id. */
+    private final String replicaGrpId;
+
+    /** Replica listener. */
+    private final ReplicaListener listener;
+
+    /**
+     * The map matches an operation id to the future of operation result.
+     * The first id is a business transaction id within which the operation is handled.
+     * The second one is an id of operation (the id is locally generated when the replication request is received).
+     * Operation future is a leaf element for waiting the operation completion and receiving a result.
+     */
+    private final ConcurrentHashMap<UUID, ConcurrentHashMap<UUID, CompletableFuture>> ops = new ConcurrentHashMap<>();
+
+    /**
+     * The constructor of replica server.
+     *
+     * @param replicaGrpId Replication group id.
+     * @param listener Replica listener.
+     */
+    public Replica(
+            String replicaGrpId,
+            ReplicaListener listener
+    ) {
+        this.replicaGrpId = replicaGrpId;
+        this.listener = listener;
+    }
+
+    /**
+     * Process a replication request on the replica.
+     *
+     * @param request Request to replication.
+     * @return Response.
+     */
+    public ReplicaResponse processRequest(ReplicaRequest request) { // define proper set of exceptions that might be thrown.
+        assert replicaGrpId.equals(request.locator().groupId()) : IgniteStringFormatter.format(
+                "Partition mismatch: request does not match the replica [reqPartId={}, replicaGrpId={}]", request.locator().groupId(),
+                replicaGrpId);
+
+        //TODO:IGNITE-17378 Check replica is alive.
+
+        ReplicaResponse response;
+
+        try {
+            if (request instanceof WaiteOperationResultRequest) {
+                var opsRequest = (WaiteOperationResultRequest) request;
```

Comment:
I believe that we may introduce private handleWaitOperatoinResultRequest(WaiteOperationResultRequest request) that besides better encapsulation will eliminate explicit cast.